### PR TITLE
:bug: :video_camera: When I select a video in the modal but opt to close the window, the video shouldn't be uploaded 

### LINKED ISSUE #486: 🐛 📹 When I select a video in the modal but opt to close the window, the video shouldn't be uploaded
**Technology and Knowledge Scope 🧠**
- [x] RxJs
- [x] HTML, SCSS
- [x] Typescript
- [x] Firebase Storage
- [x] Nx

## Describe the bug
When I select a video in the modal, but opt to close the modal rather than click "apply", the video should not be uploaded into the video block.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to  video block.
2. Open the video modal.
3. Click to upload video.
4. When video uploads, close the modal.
5. See error - video uploads anyway.

## Expected behaviour
When I close the modal rather than clicking "apply", then the modal should close and the video block should remain empty.

## Technical Description.
When a video is selected, it should be shown but not uploaded, only upload the video when the apply button is clicked.

### PR DESCRIPTION
**Pull request by Michael-Otieno**
# Description

This pull request solves issue #486, whereby I have redesigned the Block List UI following the design's guidelines.

Fixes #486

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Screenshot (optional)
[screen-capture (2).webm](https://github.com/italanta/elewa/assets/85155606/11dfcb90-f2a5-4b4f-98e6-02b9f1fd1d53)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
-  [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Michael-Otieno** [comment]
> Wait for the video to finish uploading before closing the modal.

Waiting for review

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts [hunk 1]
```diff
@@ -32,43 +34,57 @@ export class VideoUploadModalComponent implements OnInit{
...
     private dialogRef: MatDialogRef<VideoUploadModalComponent>,
     private _videoUploadService: FileStorageService,
-    @Inject(MAT_DIALOG_DATA) public data:{videoMessageForm: FormGroup},
-  ) {}
+    private fb: FormBuilder,
+    @Inject(MAT_DIALOG_DATA) public data: { videoMessageForm: FormGroup }
+  ) {
+    this.videoModalForm = this.fb.group({
+      fileName: [''],
+    });
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts:42]: No need for this. It's redundant

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts [hunk 2]
```diff
@@ -32,43 +34,57 @@ export class VideoUploadModalComponent implements OnInit{
...
       };
     }
-  } 
+  }
+
+  apply() {
+    const file = this.selectedFile;
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onload = async () => {
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-upload-modal/video-upload-modal.component.ts:69]: We only need the reader class so we can show the video. No need for this during upload.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Just a few changes left. Also check on this -> [Atomic commits](https://www.conventionalcommits.org/en/v1.0.0/)

**Comment by Michael-Otieno** [comment]
> Just a few changes left. Also check on this -> [Atomic commits](https://www.conventionalcommits.org/en/v1.0.0/), you'll find it beneficial

thanks for the resource

**Comment by LemmyMwaura** [comment]
Also, patch the name value after a video is selected if the user hasn't typed any and reset it after the modal is closed.

**Pull request merged**